testing/camunda-process-test-java/src/main/java/io/camunda/process/test/api/BpmnAssert.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.process.test.api;
+
+import io.camunda.process.test.api.assertions.ProcessInstanceAssert;
+import io.camunda.process.test.impl.assertions.CamundaDataSource;
+import io.camunda.process.test.impl.assertions.ProcessInstanceAssertj;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import io.camunda.zeebe.client.api.response.ProcessInstanceResult;
+import java.time.Duration;
+import org.awaitility.Awaitility;
+
+/**
+ * The entry point for all assertions.
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ *   &#064;Test
+ *   void shouldWork() {
+ *     // given
+ *     final ProcessInstanceEvent processInstance =
+ *         zeebeClient
+ *             .newCreateInstanceCommand()
+ *             .bpmnProcessId("process")
+ *             .latestVersion()
+ *             .send()
+ *             .join();
+ *
+ *     // when
+ *
+ *     // then
+ *     BpmnAssert.assertThat(processInstance)
+ *         .isCompleted()
+ *         .hasCompletedElements("A", "B");
+ *   }
+ * }
+ * </pre>
+ */
+public class BpmnAssert {
+
+  /** The default time how long an assertion waits until the expected state is reached. */
+  public static final Duration DEFAULT_ASSERTION_TIMEOUT = Duration.ofSeconds(10);
+
+  /** The default time between two assertion attempts until the expected state is reached. */
+  public static final Duration DEFAULT_ASSERTION_INTERVAL = Duration.ofMillis(100);
+
+  private static final ThreadLocal<CamundaDataSource> DATA_SOURCE = new ThreadLocal<>();
+
+  static {
+    Awaitility.setDefaultTimeout(DEFAULT_ASSERTION_TIMEOUT);
+    Awaitility.setDefaultPollInterval(DEFAULT_ASSERTION_INTERVAL);
+  }
+
+  // ======== Configuration options ========
+
+  /**
+   * Configures the time how long an assertion waits until the expected state is reached.
+   *
+   * @param assertionTimeout the maximum time of an assertion
+   * @see #DEFAULT_ASSERTION_TIMEOUT
+   */
+  public static void setAssertionTimeout(final Duration assertionTimeout) {
+    Awaitility.setDefaultTimeout(assertionTimeout);
+  }
+
+  /**
+   * Configures the time between two assertion attempts until the expected state is reached.
+   *
+   * @param assertionInterval time between two assertion attempts
+   * @see #DEFAULT_ASSERTION_INTERVAL
+   */
+  public static void setAssertionInterval(final Duration assertionInterval) {
+    Awaitility.setDefaultPollInterval(assertionInterval);
+  }
```

Comment:
🔧 Maybe not fully in the scope of this PR, but it would be nice to make these configurable through the extension. E.g.:

`@CamundaProcessTest(assertionTimeout=10, assertionInterval=100)`

Reply:
Interesting idea. I'm not sure about it. Currently, these properties are configured globally for all test cases. So, it doesn't feel correct to change them in one annotation. The other test cases would be accepted too. 

Let's think about it later again. :+1: